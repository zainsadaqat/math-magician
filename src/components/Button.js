import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  return (
    <button
      type="button"
      className={props.className}
      value={props.value}
      onClick={props.clickEvent}
    >
      {props.value}
    </button>
  );
};

Button.propTypes = {
  value: PropTypes.string.isRequired,
  clickEvent: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired,
};

export default Button;
