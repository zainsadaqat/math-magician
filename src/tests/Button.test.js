import renderer from 'react-test-renderer';
import Button from '../components/Button';

test('Matches Snapshot', () => {
  const tree = renderer
    .create(
      <Button />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
