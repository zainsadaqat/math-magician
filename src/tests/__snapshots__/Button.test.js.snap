// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<button
  type="button"
/>
`;
